### ISSUE TITLE
cannot load such file -- aws-sdk-efs (LoadError)

### ISSUE THREAD

**Issue by gavinmoorcroft** [inspec/inspec-aws#202]
It seems there was a recent MR completed (4 hours ago) for issue #184. Prior to this our aws tests were executing fine. After this MR was done we now get the error below when running out aws tests

Maybe coincidence but I notice the MR was to add AWS EFS support and the error is complaining about not finding 'aws-sdk-efs'

```
Traceback (most recent call last):
	34: from /usr/local/bundle/bin/inspec:23:in `<main>'
	33: from /usr/local/bundle/bin/inspec:23:in `load'
	32: from /usr/local/bundle/gems/inspec-bin-4.18.97/bin/inspec:11:in `<top (required)>'
	31: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/base_cli.rb:35:in `start'
	30: from /usr/local/bundle/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
	29: from /usr/local/bundle/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	28: from /usr/local/bundle/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	27: from /usr/local/bundle/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	26: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/cli.rb:289:in `exec'
	25: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/runner.rb:129:in `run'
	24: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/runner.rb:101:in `load'
	23: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/runner.rb:101:in `each'
	22: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/runner.rb:107:in `block in load'
	21: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `load_libraries'
	20: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `each_with_index'
	19: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `each'
	18: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:265:in `block in load_libraries'
	17: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `load_libraries'
	16: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `each_with_index'
	15: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:247:in `each'
	14: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:265:in `block in load_libraries'
	13: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile.rb:272:in `load_libraries'
	12: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:143:in `load_libraries'
	11: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:143:in `each'
	10: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:144:in `block in load_libraries'
	 9: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:159:in `load_library_file'
	 8: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:170:in `load_with_context'
	 7: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/profile_context.rb:170:in `instance_eval'
	 6: from libraries/aws_alb.rb:3:in `load_with_context'
	 5: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/dsl_shared.rb:47:in `require'
	 4: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/dsl_shared.rb:47:in `eval'
	 3: from libraries/aws_backend.rb:27:in `create'
	 2: from /usr/local/bundle/gems/inspec-core-4.18.97/lib/inspec/dsl_shared.rb:33:in `require'
	 1: from /usr/local/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/usr/local/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- aws-sdk-efs (LoadError)
```

Cheers

**Comment by rmoles** [comment]
Hi @gavinmoorcroft, 
In order to help further on this issue, would you be able to provide a bit more information on your setup?
* What version of insepc is being used?
* What version of the resource pack is being used?

**Comment by gavinmoorcroft** [comment]
hey, thanks for the speedy reply!!

inspec verison: 4.18.97

Resource pack version: not sure how to provide that, can you give me some guidance? The resource is just loaded via the master.tar.gz.

This is the inspec.yml:

`name: inspec_subnet`
`title: InSpec SUBNET`
`license: Apache-2.0`
`summary: An InSpec Compliance Profile for testing AWS SUBNET`
`version: 0.0.1`
`inspec_version: '>=4.6.9'`
`depends:`
` - name: inspec-aws`
`   url: https://github.com/inspec/inspec-aws/archive/master.tar.gz`
`supports:`
`  - platform: aws`

Sorry, I should have included that in the first post. My bad.

**Comment by rmoles** [comment]
@gavinmoorcroft Looking at the information you have provided there's a couple of issues that are causing the issue above.
I see that the URL used in the inspec.yml is pointing at `master.tar.gz`. We would recommend to avoid any instability in the future to pin to a specific version of the resource pack. Versions can be found here: https://github.com/inspec/inspec-aws/releases
The reason for this issue is down to a change in the underlying train-aws plugin. We have added the efs dependency and this new version of the train-aws plugin is only available in inspec `4.18.108` onwards.

The TLDR is. If you wish to continue using the master version of the resource pack inspec must be updated to the latest version. Although we would recommend pinning to a specific version of the resource pack and only updating whenever a new version has been verified as working in your local setup.

Let me know if this solves your issue.

**Comment by gavinmoorcroft** [comment]
Hey,

I was wondering if that was the issue. I will push us back to the version of the resource that we know what works then have a meeting with the team and decide what to do.

Thanks for your time.
Gavin.
